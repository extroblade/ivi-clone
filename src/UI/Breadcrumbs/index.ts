export * from './Breadcrumbs';
export * from './MovieBreadcrumbs';
