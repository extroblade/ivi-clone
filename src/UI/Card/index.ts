export * from './Card';
export * from './CardLoader';
export * from './ShowAll';
