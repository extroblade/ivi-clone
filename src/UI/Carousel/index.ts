export * from './Carousel';
export * from './CommentCarousel/CommentCarousel';
export * from './PromoCarousel/PromoCarousel';
export * from './SimilarMovies';
export * from './Top10Carousel/Top10Carousel';
