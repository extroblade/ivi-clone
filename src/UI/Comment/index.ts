export * from './Comment';
export * from './CommentAvatar';
export * from './CommentCard/CommentCard';
export * from './CommentInput';
export * from './CommentSection';
export * from './Vote/Vote';
