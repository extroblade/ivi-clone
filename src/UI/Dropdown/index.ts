export * from './ChooseDropdown';
export * from './Dropdown';
export * from './SearchDropdown';
