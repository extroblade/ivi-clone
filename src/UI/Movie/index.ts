export * from './AddNewMovie/AddNewMovie';
export * from './MovieBGContainer/MovieBGContainer';
export * from './MovieInfo/MovieInfo';
export * from './MovieInfo/MovieTitle';
export * from './MovieOptions/MovieOptions';
