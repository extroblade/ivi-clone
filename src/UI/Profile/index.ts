export * from './ProfileButton/ProfileButton';
export * from './ProfileIcon/ProfileIcon';
export * from './ProfileSelector/ProfileSelector';
