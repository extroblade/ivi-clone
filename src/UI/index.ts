export * from './Alert/Alert';
export * from './Alert/AlertsList';
export * from './Badge/Badge';
export * from './BarGraph/BarGraph';
export * from './Breadcrumbs';
export * from './Button/Button';
export * from './Card';
export * from './Carousel';
export * from './Comment';
export * from './Description/Description';
export * from './Dropdown';
export * from './EmptyNotifications/EmptyNotifications';
export * from './Explanations/Explanations';
export * from './FullScreenModal/FullScreenModal';
export * from './Grid/Grid';
export * from './Htag/Htag';
export * from './Input/Input';
export * from './Loader/Loader';
export * from './Movie';
export * from './NotFound/NotFound';
export * from './P/P';
export * from './Plank/InputRange';
export * from './Plank/Plank';
export * from './Player/Player';
export * from './Profile';
export * from './RatingBlock';
export * from './RegisterForm/RegisterForm';
export * from './Sausage/Sausage';
export * from './Sup/Sup';
export * from './Title/Title';
export * from './WatchAllDevices/WatchAllDevices';
