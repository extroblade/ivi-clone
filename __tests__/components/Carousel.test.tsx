import { render } from '@testing-library/react';
import React from 'react';

import { Carousel } from '@/UI';

const testArray = [1, 2, 3, 4, 5];

describe('carousel', () => {
  test('number array', () => {
    const carousel = render(<Carousel title={'title'} movies={testArray} />);
    expect(carousel).toMatchSnapshot();
  });
});
