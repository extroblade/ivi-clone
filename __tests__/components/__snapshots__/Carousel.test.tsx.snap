// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`carousel number array 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel"
      >
        <div
          class="title"
        >
          <a
            class="title_container"
            href=""
          >
            <div
              title="title"
            >
              <h4
                class="h h4"
              >
                title
              </h4>
            </div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
          </a>
        </div>
        <div
          class="slick-slider slick-initialized"
          dir="ltr"
        >
          <div
            class="arrow_container"
          >
            <div
              class="arrow prev"
            >
              <button
                class="button transparent L"
                disabled=""
              >
                <svg
                  fill="currentColor"
                  height="30"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="card"
                  >
                    <div
                      class="imageSection loader"
                    />
                    <div
                      class="textSection loader"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="card"
                  >
                    <div
                      class="imageSection loader"
                    />
                    <div
                      class="textSection loader"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="2"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="card"
                  >
                    <div
                      class="imageSection loader"
                    />
                    <div
                      class="textSection loader"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="3"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="card"
                  >
                    <div
                      class="imageSection loader"
                    />
                    <div
                      class="textSection loader"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="4"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="card"
                  >
                    <div
                      class="imageSection loader"
                    />
                    <div
                      class="textSection loader"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="5"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="card"
                  >
                    <div
                      class="imageSection loader"
                    />
                    <div
                      class="textSection loader"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="6"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="card"
                  >
                    <div
                      class="imageSection loader"
                    />
                    <div
                      class="textSection loader"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="7"
                style="outline: none; width: 0px;"
                tabindex="-1"
              />
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="8"
                style="outline: none; width: 0px;"
                tabindex="-1"
              />
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="9"
                style="outline: none; width: 0px;"
                tabindex="-1"
              />
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="10"
                style="outline: none; width: 0px;"
                tabindex="-1"
              />
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="11"
                style="outline: none; width: 0px;"
                tabindex="-1"
              />
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="12"
                style="outline: none; width: 0px;"
                tabindex="-1"
              />
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="13"
                style="outline: none; width: 0px;"
                tabindex="-1"
              />
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="14"
                style="outline: none; width: 0px;"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            class="arrow_container"
          >
            <div
              class="arrow next"
            >
              <button
                class="button transparent L"
              >
                <svg
                  fill="currentColor"
                  height="30"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel"
    >
      <div
        class="title"
      >
        <a
          class="title_container"
          href=""
        >
          <div
            title="title"
          >
            <h4
              class="h h4"
            >
              title
            </h4>
          </div>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </a>
      </div>
      <div
        class="slick-slider slick-initialized"
        dir="ltr"
      >
        <div
          class="arrow_container"
        >
          <div
            class="arrow prev"
          >
            <button
              class="button transparent L"
              disabled=""
            >
              <svg
                fill="currentColor"
                height="30"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="card"
                >
                  <div
                    class="imageSection loader"
                  />
                  <div
                    class="textSection loader"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="card"
                >
                  <div
                    class="imageSection loader"
                  />
                  <div
                    class="textSection loader"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="2"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="card"
                >
                  <div
                    class="imageSection loader"
                  />
                  <div
                    class="textSection loader"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="3"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="card"
                >
                  <div
                    class="imageSection loader"
                  />
                  <div
                    class="textSection loader"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="4"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="card"
                >
                  <div
                    class="imageSection loader"
                  />
                  <div
                    class="textSection loader"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="5"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="card"
                >
                  <div
                    class="imageSection loader"
                  />
                  <div
                    class="textSection loader"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="6"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  class="card"
                >
                  <div
                    class="imageSection loader"
                  />
                  <div
                    class="textSection loader"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="7"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="8"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="9"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="10"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="11"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="12"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="13"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="14"
              style="outline: none; width: 0px;"
              tabindex="-1"
            />
          </div>
        </div>
        <div
          class="arrow_container"
        >
          <div
            class="arrow next"
          >
            <button
              class="button transparent L"
            >
              <svg
                fill="currentColor"
                height="30"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
